src/components/header.component.vue

```diff
@@ -0,0 +1,395 @@
+<template>
+  <div class="HeaderWrapper">
+    <div class="HeaderBox">
+      <span class="LogoWrapper" v-bind:class="isShowCategory ? 'Active' : ''">
+        <a href="/">
+          <img src="../assets/CLASS101_Logo.svg" alt="/" />
+        </a>
+      </span>
+      <div class="SearchInput" v-bind:style="SearchInputHandler">
+        <form class="SearchContainer" v-bind:style="SearchContainerHandler">
+          <input
+            type="search"
+            placeholder="찾으시는 취미가 있으신가요?"
+            maxlength="100"
```

Comment:
선택사항이지만, 숫자 데이터는 변수로 따로 관리해주면 더 좋습니다. 특히 여러군데 중복되어 사용되는 숫자면 좋습니다.
그렇다고 모든 숫자를 다 분기하면 오히려 코드 흐름 읽기 안 좋으니까 변경 가능성이 있을 법한 숫자만 분리하는게 중요합니다.
저희도 변경될 가능성 있는 숫자도 그냥 쓰기도 하니까 선택적으로 수용하시면 됩니다

```suggestion
            :maxlength="searchMaxLength"
            .
            .
            data() {
              return {
                searchMaxLength: 100
              }
            },
```